Fix unstable `invalid block application` test

Test sometimes fails in Travis:
```
apply invalid full block - State: utxo, Verify Transactions: true, PoPoW Bootstrap: false *** FAILED ***
java.lang.AssertionError: assertion failed: timeout (6 seconds) during expectMsgClass waiting for interface scorex.core.network.NodeViewSynchronizer$ReceivableMessages$ModificationOutcome[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.Predef$.assert(Predef.scala:219)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:509)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:482)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:482)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at akka.testkit.TestKit.expectMsgType(TestKit.scala:896)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.ergoplatform.utils.NodeViewBaseOps.expectMsgType(NodeViewTestOps.scala:100)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.ergoplatform.utils.NodeViewBaseOps.expectMsgType$(NodeViewTestOps.scala:100)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.ergoplatform.nodeView.ErgoNodeViewHolderSpec.expectMsgType(ErgoNodeViewHolderSpec.scala:20)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.ergoplatform.utils.NodeViewBaseOps.expectModificationOutcome(NodeViewTestOps.scala:71)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.ergoplatform.utils.NodeViewBaseOps.expectModificationOutcome$(NodeViewTestOps.scala:70)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  ...[0m[0m
```

--- a/src/test/scala/org/ergoplatform/utils/generators/ErgoTransactionGenerators.scala
+++ b/src/test/scala/org/ergoplatform/utils/generators/ErgoTransactionGenerators.scala
@@ -111,6 +111,7 @@ trait ErgoTransactionGenerators extends ErgoGenerators {
     * Generated transaction may still be invalid, if:
     * - default prover does not know how to sign at least one input
     * - number of assets exceeds Transaction.MaxTokens
+    * - transaction is trying to create dust
     */
   def validTransactionFromBoxes(boxesToSpend: IndexedSeq[ErgoBox],
                                 rnd: Random = new Random,

--- a/src/test/scala/org/ergoplatform/utils/generators/ValidBlocksGenerators.scala
+++ b/src/test/scala/org/ergoplatform/utils/generators/ValidBlocksGenerators.scala
@@ -11,6 +11,7 @@ import org.ergoplatform.modifiers.mempool.{ErgoTransaction, UnsignedErgoTransact
 import org.ergoplatform.nodeView.state._
 import org.ergoplatform.nodeView.state.wrapped.WrappedUtxoState
 import org.ergoplatform.settings.{Algos, Constants, ErgoSettings}
+import org.ergoplatform.utils.LoggingUtil
 import org.ergoplatform.{ErgoBox, ErgoBoxCandidate, Input}
 import org.scalatest.Matchers
 import scorex.core.VersionTag
@@ -21,7 +22,7 @@ import sigmastate.Values
 import sigmastate.interpreter.{ContextExtension, ProverResult}
 
 import scala.annotation.tailrec
-import scala.util.{Random, Try}
+import scala.util.{Failure, Random, Success, Try}
 
 trait ValidBlocksGenerators
   extends TestkitHelpers with FileUtils with Matchers with ChainGenerator with ErgoTransactionGenerators {
@@ -78,12 +79,24 @@ trait ValidBlocksGenerators
             val (consumedBoxesFromState, remainedBoxes) = stateBoxes.splitAt(Try(rnd.nextInt(stateBoxes.size) + 1).getOrElse(0))
             // disable tokens generation to avoid situation with too many tokens
             val tx = validTransactionFromBoxes((consumedSelfBoxes ++ consumedBoxesFromState).toIndexedSeq, rnd, issueNew)
-            loop(remainedBoxes, remainedSelfBoxes ++ tx.outputs, tx +: acc, rnd)
+            tx.statelessValidity match {
+              case Failure(e) =>
+                log.warn(s"Failed to generate valid transaction: ${LoggingUtil.getReasonMsg(e)}")
+                loop(stateBoxes, selfBoxes, acc, rnd)
+              case _ =>
+                loop(remainedBoxes, remainedSelfBoxes ++ tx.outputs, tx +: acc, rnd)
+            }
           } else {
             // take all remaining boxes from state and return transactions set
             val (consumedSelfBoxes, remainedSelfBoxes) = selfBoxes.splitAt(1)
             val tx = validTransactionFromBoxes((consumedSelfBoxes ++ stateBoxes).toIndexedSeq, rnd, issueNew)
-            ((tx +: acc).reverse, remainedSelfBoxes ++ tx.outputs ++ createdEmissionBox)
+            tx.statelessValidity match {
+              case Failure(e) =>
+                log.warn(s"Failed to generate valid transaction: ${LoggingUtil.getReasonMsg(e)}")
+                loop(stateBoxes, selfBoxes, acc, rnd)
+              case _ =>
+                ((tx +: acc).reverse, remainedSelfBoxes ++ tx.outputs ++ createdEmissionBox)
+            }
           }
       }
     }

